src/main/java/org/candlepin/subscriptions/resource/HostsResource.java

```diff
@@ -116,15 +116,42 @@ else if (uom == Uom.SOCKETS) {
             displayNameContains :
             "";
         Pageable page = ResourceUtils.getPageable(offset, limit, sortValue);
-        Page<TallyHostView> hosts = repository.getTallyHostViews(
-            accountNumber,
-            productId.toString(),
-            sanitizedSla,
-            sanitizedUsage,
-            sanitizedDisplayNameSubstring,
-            minCores,
-            minSockets,
-            page);
+
+        boolean isSpecial = Objects.equals(productId, ProductId.OPENSHIFT_DEDICATED_METRICS) ||
+            Objects.equals(productId, ProductId.OPENSHIFT_METRICS);
+
+        List<org.candlepin.subscriptions.utilization.api.model.Host> payload;
+        Page<?> hosts;
+        if (isSpecial) {
+
+            OffsetDateTime start = OffsetDateTime.now();
+            OffsetDateTime end = OffsetDateTime.now();
+
+            if (Objects.isNull(beginning) && Objects.nonNull(ending)) {
+                start = ending;
+            }
+
+            if (Objects.isNull(ending) && Objects.nonNull(beginning)) {
+                end = beginning;
+            }
```

Comment:
There's a subtle issue here: when you pass both beginning and ending, start and date get set to `OffsetDateTime.now()`.

How about:

```suggestion
            OffsetDateTime now = OffsetDateTime.now();
            OffsetDateTime start = Optional.of(beginning).orElse(now);
            OffsetDateTime end = Optional.of(ending).orElse(now);
```
instead ?